oap-server/server-storage-plugin/storage-shardingsphere-plugin/src/test/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/shardingsphere/ITShardingTest.java

```diff
@@ -0,0 +1,732 @@
+        propertiesDs0.setProperty("dataSource.password", "root@1234");
+
+        String dsUrl1 = "jdbc:" + driverType + "://" +
+            ENVIRONMENT.getServiceHost("data-source-1", dsServicePort) + ":" +
+            ENVIRONMENT.getServicePort("data-source-1", dsServicePort) +
+            urlSuffix;
+        Properties propertiesDs1 = new Properties();
+        propertiesDs1.setProperty("jdbcUrl", dsUrl1);
+        propertiesDs1.setProperty("dataSource.user", dsUserName);
+        propertiesDs1.setProperty("dataSource.password", dsPassword);
+
+        ssClient = new JDBCHikariCPClient(properties);
+        dsClient0 = new JDBCHikariCPClient(propertiesDs0);
+        dsClient1 = new JDBCHikariCPClient(propertiesDs1);
+
+        ssClient.connect();
+        dsClient0.connect();
+        dsClient1.connect();
+    }
+
+    private void initTestData() {
+        moduleManager = mock(ModuleManager.class);
+        ConfigService configService = mock(ConfigService.class);
+        Whitebox.setInternalState(moduleManager, "isInPrepareStage", false);
+        when(configService.getMetricsDataTTL()).thenReturn(ttlTestCreate);
+        when(configService.getRecordDataTTL()).thenReturn(ttlTestCreate);
+        when(configService.getSearchableTracesTags()).thenReturn(searchableTag);
+        when(moduleManager.find(anyString())).thenReturn(mock(ModuleProviderHolder.class));
+        when(moduleManager.find(CoreModule.NAME).provider()).thenReturn(mock(ModuleServiceHolder.class));
+        when(moduleManager.find(CoreModule.NAME).provider().getService(ConfigService.class))
+            .thenReturn(configService);
+        mySQLTableInstaller = new MySQLTableInstaller(ssClient, moduleManager);
+        installer = new ShardingSphereTableInstaller(ssClient, moduleManager,
+                                                     dataSources,
+                                                     mySQLTableInstaller
+        );
+
+        durationWithinTTL.setConfigService(configService);
+
+        duration = new Duration();
+        duration.setStart(DateTime.now().minusMinutes(15).toString("yyyy-MM-dd HHmm"));
+        duration.setEnd(DateTime.now().plusMinutes(1).toString("yyyy-MM-dd HHmm"));
+        duration.setStep(Step.MINUTE);
+
+        entityA = new Entity();
+        entityA.setServiceName("Service_A");
+        entityA.setNormal(true);
+        entityA.setScope(Scope.Service);
+
+        entityB = new Entity();
+        entityB.setServiceName("Service_B");
+        entityB.setNormal(true);
+        entityB.setScope(Scope.Service);
+
+        serviceIdA = entityA.buildId();
+        serviceIdB = entityB.buildId();
+    }
+
+    @SneakyThrows
+    @After
+    public void after() {
+        ENVIRONMENT.stop();
+    }
+
+    @SneakyThrows
+    @Test
+    public void test() {
+        trafficTest();
+        metricsTest();
+        tagsTest();
+        recordsTest();
+        topologyTest();
```

Comment:
You can just annotate each of these methods with `@Test` and remove this `test()` method.

Reply:
> You can just annotate each of these methods with `@Test` and remove this `test()` method. So when any of these tests failed, you can directly see the result of failed method

NVM, adding `@Test` to each method will cause the environment to start and stop again and again